R/stan_module.R

```diff
@@ -0,0 +1,86 @@
+#' StanAll class object
+#' @slot functions Character, the functions part of a stan model
+#' @slot data Character, the data part of a stan model
+#' @slot parameters Character, the parameters part of a stan model
+#' @slot transformed_parameters Character, the transformed_parameters part of a stan model
+#' @slot model Character, the model part of a stan model
+#' @slot prior List, the priors part of a stan model
+#' @slot generated_quantities Character, the generated_quantities part of a stan model
+#' @slot includes Character
+#' @slot inits List with the initial values of the stan model.
+#' @exportClass StanModule
+StanModule <- setClass(
+    "StanModule",
+    representation(
+        functions = "character",
+        data = "character",
+        parameters = "character",
+        transformed_parameters = "character",
+        prior = "list",
+        generated_quantities = "character",
+        inits = "list"
+    )
+)
+
+
+
+#' source_stan_part exports a .stan file as a character
+#' @param filename The name of the .stan file in the directory of the package.
+#' @export
+source_stan_part <- function(filename) {
+    absolute_filename <- system.file(
+        "stanparts",
+        filename,
+        package = "jmpost"
+    )
+    readLines(absolute_filename)
+}
+
+
+#' @importFrom assertthat assert_that
+#' @param functions TODO
+#' @param data TODO
+#' @param parameters TODO
+#' @param transformed_parameters TODO
+#' @param prior TODO
+#' @param generated_quantities TODO
+#' @param inits TODO
+#' @export
+stan_module <- function(functions, data, parameters, transformed_parameters,
```

Comment:
Instead of an explicitly function I think instead we should be using the "initialise" method. This is a magic method that is automatically called when users run the `StanModules()` function